### PR TITLE
#13290. Return 3rd party shared quota used

### PR DESCRIPTION
**Pull request by polmr**
**Risk area/s:**
- Nothing: no existing functionality has been modified per se

### CODE REVIEW

**Comment by sergiohs84** [comment]
Wait for the API team's input in order to check whether the `scycle` is obsolete ad to add documentation for `sgw` and `rtt`

File: include/mega/account.h [hunk 1]
```diff
@@ -84,31 +84,30 @@ struct MEGA_API AccountDetails
...
-    m_off_t storage_used = 0, storage_max = 0;
-    m_off_t transfer_own_used = 0, transfer_srv_used = 0, transfer_max = 0;
-    m_off_t transfer_own_reserved = 0, transfer_srv_reserved = 0;
+    m_off_t storage_used = 0;
+    m_off_t storage_max = 0;
+
+    // Own user transfer
+    m_off_t transfer_max = 0;
+    m_off_t transfer_own_used = 0;
+    m_off_t transfer_srv_used = 0;  // 3rd party shared quota used
```
Comment (polmr) [inline include/mega/account.h]: This could read `3rd party served quota to other users`?

File: include/mega/account.h [hunk 2]
```diff
@@ -84,31 +84,30 @@ struct MEGA_API AccountDetails
...
+    m_time_t transfer_hist_starttime = 0;   // transfer history start timestamp
+    m_time_t transfer_hist_interval = 3600; // timespan that a single transfer window record covers
+    vector<m_off_t> transfer_hist; // transfer window - oldest to newest, bytes consumed per time interval
+    bool transfer_hist_valid = true;    // transfer hist valid for overquota accounts
 
-    bool transfer_hist_valid = true;    // transfer hist valid for overquota
-                                        // accounts
+    // Reserved transfer quota for ongoing transfers (currently ignored by clients)
+    m_off_t transfer_reserved = 0; // free IP-based
+    m_off_t transfer_srv_reserved = 0; // 3rd party
```
Comment (polmr) [inline include/mega/account.h]: `3rd party served` ?

Comment (sergiohs84) [reply include/mega/account.h]: It is actually reserved, but not served yet :stuck_out_tongue_closed_eyes:

File: include/megaapi.h [hunk 3]
```diff
@@ -15893,11 +15893,26 @@ class MegaAccountDetails
...
     /**
-     * @brief Get the used bandwidth
+     * @brief Get the used bandwidth for own user allowance
+     * @see: MegaAccountDetails::getTransferUsed
      * @return Used bandwidth (in bytes)
      */
     virtual long long getTransferOwnUsed();
 
+    /**
+     * @brief Get the used bandwidth from 3rd party shared quota
```
Comment (sergiohs84) [inline include/megaapi.h]:
```suggestion
     * @brief Get the used bandwidth served to other users
```

File: include/megaapi.h [hunk 4]
```diff
@@ -15893,11 +15893,26 @@ class MegaAccountDetails
...
+     * @see: MegaAccountDetails::getTransferUsed
+     * @return Used bandwidth (in bytes)
+     */
+    virtual long long getTransferSrvUsed();
+
+    /**
+     * @brief Get the used bandwidth including own, free & 3rd party allowance
```
Comment (sergiohs84) [inline include/megaapi.h]:
```suggestion
     * @brief Get the used bandwidth allowance including own, free and served to other users
```

File: src/commands.cpp [hunk 5]
```diff
@@ -4055,38 +4048,48 @@ void CommandGetUserQuota::procresult()
...
                 break;
 
             case MAKENAMEID6('c', 's', 'x', 'f', 'e', 'r'):
-                // third-party transfer quota used
+            // PRO transfer quota served to others
+                details->transfer_srv_used += client->json.getint();
+                break;
+
+            case MAKENAMEID3('t', 'u', 'a'):
+            // Transfer usage by 3rd parties (actor) which hasn't yet been committed back to the API DB. Supplements csxfer
```
Comment (sergiohs84) [inline src/commands.cpp]:
```suggestion
            // Transfer usage served to other users which hasn't yet been committed back to the API DB. Supplements csxfer
```

**Pull request merged**